providers/edge/src/airflow/providers/edge/plugins/templates/edge_worker_hosts.html

```diff
@@ -100,30 +108,50 @@ <h2>Edge Worker Hosts</h2>
+            {%- if host.state in ["idle", "running"] -%}
+              <td>
+                <button id="button_{{ host.worker_name }}" onclick="showform('{{ host.worker_name }}')" style="padding: 10px 20px; background-color: blue; color: white; border: none; border-radius: 5px;">
+                    Enter Maintenance
+                </button>
+                <form id="form_{{ host.worker_name }}" style="display: none" action="../edgeworkerhosts/status/maintenance/{{ host.worker_name }}/on" method="POST">
                   <input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
+                  <div style="padding-bottom: 1em;">
+                    <label for="maintenance_comment">Maintenance Comment:</label>
+                  </div>
+                  <textarea  name="maintenance_comment" rows="3" maxlength="1024" required></textarea>
+                  <br />
                   <button type="submit" style="padding: 10px 20px; background-color: blue; color: white; border: none; border-radius: 5px;">
-                      Enter
+                      Confirm Maintenance
                   </button>
-                </form>
-              {%- elif host.state in ["maintenance pending", "maintenance mode", "maintenance request"] -%}
-                <form action="../edgeworkerhosts/status/maintenance/{{ host.worker_name }}/off" method="POST">
+                </td>
+              </form>
+            {%- elif host.state in ["maintenance pending", "maintenance mode", "maintenance request"] -%}
+              <form action="../edgeworkerhosts/status/maintenance/{{ host.worker_name }}/off" method="POST">
+                <td style="text-align: center; vertical-align: middle;">
+                  <textarea  name="maintenance_comment" rows="3" maxlength="1024" style="display: block; margin: 0 auto; width:100%;" required>{{ host.maintenance_comment }}</textarea>
                   <input type="hidden" name="csrf_token" value="{{ csrf_token() }}"/>
-                  <button type="submit" style="padding: 10px 20px; background-color: blue; color: white; border: none; border-radius: 5px;">
-                      Exit
-                  </button>
-                </form>
-              {%- elif host.state in ["offline", "unknown", "offline maintenance"] -%}
-                <form action="../edgeworkerhosts/status/maintenance/{{ host.worker_name }}/remove" method="POST">
+                  <div style="margin-top: 10px;">
+                    <button type="submit" style="padding: 10px 20px; background-color: blue; color: white; border: none; border-radius: 5px; margin-right: 5px;">
+                        Exit Maintenance
+                    </button>
+                    <button type="submit" style="padding: 10px 20px; background-color: blue; color: white; border: none; border-radius: 5px;" formaction="../edgeworkerhosts/status/maintenance/{{ host.worker_name }}/change_comment">
```

Comment:
Functionally this works as described. But the layout of the form is not really beautiful. Can you maybe adjust the buttons to be similar like on other admin pages and beautify the layout a bit?
I know the UI of Airflow 2 will go away soon, but it should be a bit more consistent.